### PR TITLE
force delete the global hub addons when waiting for the deletion timeout

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: myan <myan@redhat.com>
Resolves: https://issues.redhat.com/browse/ACM-5029

- Add force deletion of the global hub addons 
- Prune job only removes the global hub finalizer from the global resources

### CODE REVIEW

File: operator/pkg/controllers/hubofhubs/globalhub_prune.go [hunk 1]
```diff
@@ -166,27 +166,82 @@ func (r *MulticlusterGlobalHubReconciler) deleteClusterManagementAddon(ctx conte
 	return nil
 }
 
-func (r *MulticlusterGlobalHubReconciler) waitUtilAddonDeleted(ctx context.Context, log logr.Logger) error {
-	ctx, cancel := context.WithTimeout(ctx, 60*time.Second)
+// delete the managedclusteraddons with graceful duration, if reach out of the graceful duration,
+// then force delete them.
+func (r *MulticlusterGlobalHubReconciler) waitUtilAddonDeleted(ctx context.Context, log logr.Logger,
```
Comment (clyang82) [inline operator/pkg/controllers/hubofhubs/globalhub_prune.go:171]: I remembered addon framework provide this capability. Do we really need to handle the force deletion in our logic?

Comment (clyang82) [reply operator/pkg/controllers/hubofhubs/globalhub_prune.go:171]: @zhiweiyin318 ^

Comment (zhiweiyin318) [reply operator/pkg/controllers/hubofhubs/globalhub_prune.go:171]: import-controller will force delete all addon CRs and manifestWorks after the cluster is unavailable.

**Comment by yanmxa** [comment]
Close this PR Cause we don't need to add the force deletion, instead to diagnose the reason why it can't be deleted.

**Pull request closed without merging**